### PR TITLE
client: use fh min readahead to enable readahead

### PR DESCRIPTION
**Pull request by batrick**
The client was using the configured maximum readahead of 0 which was changed in
95ee699. This prevented readahead from ever
running (without setting a different default from 0).

Resolves #16024.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
The code looks good to me.
@batrick, please document what testing you've done to verify this.

We also want one of our nightly tests to verify that readahead is actually happening — we didn't notice it was broken previously!

File: src/osdc/Striper.cc [hunk 1]
```diff
@@ -244,7 +244,7 @@ uint64_t Striper::get_num_objects(const file_layout_t& layout,
   __u32 object_size = layout.object_size;
```
Comment (gregsfortytwo) [inline src/osdc/Striper.cc]: Build warning!

> osdc/Striper.cc:244:9: warning: unused variable 'object_size' [-Wunused-variable]

Comment (batrick) [inline src/osdc/Striper.cc]: This is the one I told you about which is fixed in master. Should I rebase?

Comment (batrick) [inline src/osdc/Striper.cc]: Nevermind, fixed.

**Comment by gregsfortytwo** [comment]
retest this please

File: src/client/Client.h [hunk 2]
```diff
@@ -709,7 +709,8 @@ class Client : public Dispatcher, public md_config_obs_t {
   struct C_Readahead : public Context {
     Client *client;
     Fh *f;
-    C_Readahead(Client *c, Fh *f);
+    C_Readahead(Client *c, Fh *f) : client{c}, f{f} {f->get(); f->readahead.inc_pending();};
+    ~C_Readahead() {f->readahead.dec_pending(); client->_put_fh(f);};
```
Comment (gregsfortytwo) [inline src/client/Client.h]: Drop the spurious semi-colons please, and don't try to squash these into a single line — you can do a single-statement one-line function if you want, but not multiple statements like this.

**Comment by gregsfortytwo** [comment]
http://pulpito.ceph.com/gregf-2016-06-10_19:20:53-fs-greg-fs-testing-610---basic-mira/
All failures attributed to AsyncMessenger incompatibilities.

**Pull request merged**